### PR TITLE
replace JUnitParams with junit-dataprovider

### PR DESCRIPTION
**Pull request by PascalSchumacher**
see discussion at: https://github.com/joel-costigliola/assertj-core/pull/535

### CODE REVIEW

File: src/test/java/org/assertj/core/error/DescriptionFormatter_format_Test.java [hunk 1]
```diff
@@ -46,15 +46,20 @@ public void should_format_description_if_value_is_not_empty_or_null() {
...
   public void should_return_empty_String(TestDescription testDescription) {
     assertThat(formatter.format(testDescription)).isEmpty();
   }
 
-  @SuppressWarnings("unused")
-  private Object testDescriptionGenerator() {
-    return array(null,
-                 new TestDescription(null),
-                 new TestDescription(""));
+  @DataProvider(format="%m[%i]") // Workaround with custom format because TestDescription#toString returns the value
```
Comment (PascalSchumacher) [inline src/test/java/org/assertj/core/error/DescriptionFormatter_format_Test.java]: I will create a pull request based on the feedback I receive at https://github.com/TNG/junit-dataprovider/issues/66

**Comment by joel-costigliola** [comment]
Is this PR waiting for a release of JUnit DataProvider with the fix ?

**Comment by PascalSchumacher** [comment]
The PR is ready for merging. I will submit a new PR which removes the ´format="%m[%i]"´ workaround once there is a new release of junit-dataprovider (should happen in a few weeks).

**Comment by joel-costigliola** [comment]
:+1:

**Comment by joel-costigliola** [comment]
Integrated, thanks

**Pull request closed without merging**